### PR TITLE
several fixes + improvements

### PR DESCRIPTION
**Pull request by rhaschke**
This PR contains several fixes and improvements I made to suit my needs. Please decide on your own whether to consider the individual commits or not.
1. fix / work-around #13 
2. progress-bar, addresses #19
3. store expanded `.odp` in `/tmp` (only works for Linux)
This avoids cluttering my workspace with `*-expanded.odp` files.

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks a lot for your great contribution, LGTM.

**Comment by monperrus** [comment]
Thanks a lot!

See packaged file at https://github.com/monperrus/ExpandAnimations/blob/master/dist/ExpandAnimations-0.5.oxt

**Comment by katz-hm** [comment]
Well, if discontinuing Windows support was intentional, I can live with my local fix, but I wouldn't assume that “only works for Linux“ means “breaks compatibility with Windows“. 0.5 is effectively not working at all under Windows. At least it would make sense (to me) to add a note to the dist folder.

File: src/ExpandAnimations.bas [hunk 1]
```diff
@@ -73,39 +88,24 @@ function renameAsExpanded(doc as Object)
...
 
   sDocUrl = doc.getURL()
-  sDocPath = DirectoryNameoutofPath(sDocUrl, "/")
-  sDocFileName = FileNameoutofPath(sDocUrl, "/")
-  sDocFileNameExtension = GetFileNameExtension(sDocUrl)
   sDocFileNameWithoutExtension = GetFileNameWithoutExtension(sDocUrl, "/") 
-  
-  newUrlExpanded = sDocPath + "/" + sDocFileNameWithoutExtension + "-expanded.odp"
+
+  newUrlExpanded = "file:///tmp/" + sDocFileNameWithoutExtension + "-expanded.odp"
```
Comment (rhaschke) [inline src/ExpandAnimations.bas]: @katz-hm, as you seem to be a Windows user (and developer?), could you file a PR to replace the hard-coded `/tmp` here with the content of an appropriate environment variable?
According to this official document, `%TEMP%` or `%TMP%` should be suitable candidates.
Only, this variable is not set, the code should fallback to the (Linux) default `/tmp`

Comment (katz-hm) [reply src/ExpandAnimations.bas]: I'd love to contribute, but I completely lack BASIC skills. I tried reading the TEMP Env, but I think I need to replace backslashes by regular slashes it to form a valid file URI (not sure if that's all):

```
tempPath = Environ("TEMP")
newUrlExpanded = "file:///"+tempPath+"/" + sDocFileNameWithoutExtension + "-expanded.odp"
```

Resulting exception was

Message: SfxBaseModel::impl_store <file:///C:\Users\KATZ~1.BAS\AppData\Local\Temp/DijkstraAnimated-expanded.odp>

If I hardcode 
```
"file:///C:/Users/KATZ~1.BAS/AppData/Local/Temp/"+ sDocFileNameWithoutExtension + "-expanded.odp"
```
It works well. So if you could provide me with any source for the BASIC I need (or a solution to search and replace), I could prepare a PR, given that I find a way to write conditional code for the fallback, too.

Really proper way would be to additionally check for accessibility and fall back to the local folder, I think.

Comment (rhaschke) [reply src/ExpandAnimations.bas]: I can try later this evening not prepare a PR.

Comment (katz-hm) [reply src/ExpandAnimations.bas]: I'd be happy to test it.

**Pull request merged**